File: clients/JavaScript/model/camera.js
const Vec2Float = require('./vec2-float');
class Camera {
    constructor(center, rotation, attack, distance, perspective) {
        this.center = center;
        this.rotation = rotation;
        this.attack = attack;
        this.distance = distance;
        this.perspective = perspective;
    }
    static async readFrom(stream) {
        let center;
        center = await Vec2Float.readFrom(stream);
        let rotation;
        rotation = await stream.readFloat();
        let attack;
        attack = await stream.readFloat();
        let distance;
        distance = await stream.readFloat();
        let perspective;
        perspective = await stream.readBool();
        return new Camera(center, rotation, attack, distance, perspective);
    }
    async writeTo(stream) {
        let center = this.center;
        await center.writeTo(stream);
        let rotation = this.rotation;
        await stream.writeFloat(rotation);
        let attack = this.attack;
        await stream.writeFloat(attack);
        let distance = this.distance;
        await stream.writeFloat(distance);
        let perspective = this.perspective;
        await stream.writeBool(perspective);
    }
}
module.exports = Camera

File: clients/Python/model/camera.py
from .vec2_float import Vec2Float
class Camera:
    def __init__(self, center, rotation, attack, distance, perspective):
        self.center = center
        self.rotation = rotation
        self.attack = attack
        self.distance = distance
        self.perspective = perspective
    @staticmethod
    def read_from(stream):
        center = Vec2Float.read_from(stream)
        rotation = stream.read_float()
        attack = stream.read_float()
        distance = stream.read_float()
        perspective = stream.read_bool()
        return Camera(center, rotation, attack, distance, perspective)
    def write_to(self, stream):
        self.center.write_to(stream)
        stream.write_float(self.rotation)
        stream.write_float(self.attack)
        stream.write_float(self.distance)
        stream.write_bool(self.perspective)
    def __repr__(self):
        return "Camera(" + \
            repr(self.center) + "," + \
            repr(self.rotation) + "," + \
            repr(self.attack) + "," + \
            repr(self.distance) + "," + \
            repr(self.perspective) + \
            ")"
